stdlib/mime/src/main/java/org/ballerinalang/mime/util/MultipartDataSource.java

```diff
@@ -111,6 +111,12 @@ private void serializeBodyPart(OutputStream outputStream, String parentBoundaryS
             writeFinalBoundaryString(writer, parentBoundaryString);
         } catch (IOException e) {
             log.error("Error occurred while writing body parts to outputstream", e.getMessage());
+        } finally {
```

Comment:
No need for the `finally` block and explicit `close()` method call here, if we add the new `writer` object creation to the try block (by using the try-with-resources construct)

Reply:
Fixed via [83de62e](https://github.com/ballerina-platform/ballerina-lang/pull/40263/commits/83de62e8ebbab0c106a2a117fd7cd5b039e81317)